core/test/unit/middleware/spam-prevention_spec.js

```diff
@@ -19,6 +20,7 @@ describe('Middleware: spamPrevention', function () {
         spyNext = sinon.spy(function (param) {
             error = param;
         });
+        middleware.spamPrevention = rewire('../../../server/middleware/spam-prevention').__get__('spamPrevention');
```

Comment:
Yeah, that's exactly what I'm trying to achieve. 

But I don't think `spam-prevention_spec.js` should know anything about `spamPrevention`'s internel implementation. Theoretical in BDD, test should be written before implementation , and test should work regardless subject's internal refacting.

That's why I think a simple resetting by `rewire spamPrevention` sounds a good idea.